# Settings for ros_control control loop
comau_hardware_control_loop:
  loop_hz: &loop_hz 1
  cycle_time_error_threshold: 0.01

# Settings for ros_control hardware interface
comau_hardware_interface:
   use_state_server: true
   use_motion_server: true
   joints: &robot_joints
     - joint_1
     - joint_2
     - joint_3
     - joint_4
     - joint_5
     - joint_6

# Publish all joint states ----------------------------------
joint_state_controller:
   type:         joint_state_controller/JointStateController
   publish_rate: *loop_hz

# Controller for sensor tracking
sensor_tracking_controller:
   type: comau_controllers/SensorTrackingController
   joints: *robot_joints
   dead_man_timeout: 0.2
   ee_vel_limit: 5.0
   with_plot: true
   stop_trajectory_duration: 0.5
   state_publish_rate: 50 #25
   action_monitor_rate: 20

# Joint Trajectory Controller - position based -------------------------------
# For detailed explanations of parameter see http://wiki.ros.org/joint_trajectory_controller
pos_joint_traj_controller:
   type: position_controllers/JointTrajectoryController
   joints: *robot_joints
   constraints:
      goal_time: 10.6
      stopped_velocity_tolerance: 0.05
      joint_1: {trajectory: 0.2, goal: 0.1}
      joint_2: {trajectory: 0.2, goal: 0.1}
      joint_3: {trajectory: 0.2, goal: 0.1}
      joint_4: {trajectory: 0.2, goal: 0.1}
      joint_5: {trajectory: 0.2, goal: 0.1}
      joint_6: {trajectory: 0.2, goal: 0.1}
   stop_trajectory_duration: 0.5
   state_publish_rate: *loop_hz  # Frequency (in Hz) at which the controller state is published.
   action_monitor_rate: 20 # Frequency (in Hz) at which the action goal status is monitored. This is an advanced parameter that should not require changing.



